x-pack/legacy/plugins/siem/public/containers/ip_overview/index.tsx

```diff
@@ -49,7 +49,7 @@ const IpOverviewComponentQuery = React.memo<IpOverviewProps & IpOverviewReduxPro
         sourceId,
         filterQuery: createFilter(filterQuery),
         ip,
-        defaultIndex: chrome.getUiSettingsClient().get(DEFAULT_INDEX_KEY),
+        defaultIndex: useUiSetting<string[]>(DEFAULT_INDEX_KEY),
```

Comment:
Are there two different ways of doing this?

I see both:
```ts
defaultIndex: kibana.services.uiSettings.get<string[]>(DEFAULT_INDEX_KEY),
```

and
```ts
useUiSetting<string[]>(DEFAULT_INDEX_KEY),
```

Reply:
Looks like maybe one is a React Hook function and the other is just a plain API call for non-tsx React code?

Reply:
They both reference the same context provider under the hood! The former's an HOC prop (via `withKibana`) used for class components, and the latter is a hook used by function components.

Reply:
@FrankHassanabad you bring up a good point, though: I'm trying to pull as much of this as possible from the react context and avoiding grabbing state (e.g. `chrome.getUiSettingsClient`) outside of react/platform's control.